packages/strapi-plugin-i18n/config/functions/migrations/field/migrateForBookshelf.js

```diff
@@ -0,0 +1,182 @@
+    });
+  });
+  await trx.batchInsert(TMP_TABLE_NAME, tmpEntries, 100);
+};
+
+const batchUpdate = async (updatesInfo, trx, model) => {
+  const promises = updatesInfo.map(({ entriesIdsToUpdate, attributesValues }) =>
+    trx
+      .from(model.collectionName)
+      .update(attributesValues)
+      .whereIn('id', entriesIdsToUpdate)
+  );
+  await Promise.all(promises);
+};
+
+const updateFromTmpTable = async ({ model, trx, attributesToMigrate }) => {
+  const collectionName = model.collectionName;
+  if (model.client === 'pg') {
+    const substitutes = attributesToMigrate.map(() => '?? = ??.??').join(',');
+    const bindings = [collectionName];
+    attributesToMigrate.forEach(attr => bindings.push(attr, TMP_TABLE_NAME, attr));
+    bindings.push(TMP_TABLE_NAME, collectionName, TMP_TABLE_NAME);
+
+    await trx.raw(`UPDATE ?? SET ${substitutes} FROM ?? WHERE ??.id = ??.id;`, bindings);
+  } else if (model.client === 'mysql') {
+    const substitutes = attributesToMigrate.map(() => '??.?? = ??.??').join(',');
+    const bindings = [collectionName, TMP_TABLE_NAME, collectionName, TMP_TABLE_NAME];
+    attributesToMigrate.forEach(attr => bindings.push(collectionName, attr, TMP_TABLE_NAME, attr));
+
+    await trx.raw(`UPDATE ?? JOIN ?? ON ??.id = ??.id SET ${substitutes};`, bindings);
+  }
+};
+
+const createTmpTable = async ({ ORM, attributesToMigrate, model }) => {
+  const columnsToCopy = ['id', ...attributesToMigrate];
+  await ORM.knex.schema.dropTableIfExists(TMP_TABLE_NAME);
+  await ORM.knex.raw(`CREATE TABLE ?? AS ??`, [
+    TMP_TABLE_NAME,
+    ORM.knex
+      .select(columnsToCopy)
+      .from(model.collectionName)
+      .whereRaw('?', 0),
+  ]);
+};
+
+const deleteTmpTable = ({ ORM }) => ORM.knex.schema.dropTableIfExists(TMP_TABLE_NAME);
+
+const getSortedLocales = async ORM => {
+  let defaultLocale;
+  try {
+    const defaultLocaleRow = await ORM.knex('core_store')
+      .select('value')
+      .where({ key: 'plugin_i18n_default_locale' });
+    defaultLocale = defaultLocaleRow[0].value;
+  } catch (e) {
+    throw new Error("Could not migrate because the default locale doesn't exist");
+  }
+
+  let locales;
+  try {
+    locales = await ORM.knex(strapi.plugins.i18n.models.locale.collectionName).select('code');
+  } catch (e) {
+    throw new Error('Could not migrate because no locale exist');
+  }
+
+  locales.forEach(locale => (locale.isDefault = locale.code === defaultLocale));
+  return orderBy(['isDefault', 'code'], ['desc', 'asc'])(locales); // Put default locale first
+};
+
+const migrateForBookshelf = async ({ ORM, model, attributesToMigrate }) => {
+  const locales = await getSortedLocales(ORM);
+  const localizationAssoc = model.associations.find(a => a.alias === 'localizations');
+  const localizationTableName = localizationAssoc.tableCollectionName;
+
+  // The migration is custom for pg and mysql for better perfomance
+  const isPgOrMysql = ['pg', 'mysql'].includes(model.client);
```

Comment:
you could split the code into functions so you don't check the isPgOrMysql var so much